File: ITapply/ITapply.Models/Requests/EmployerUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.Requests
{
    public class EmployerUpdateRequest
    {
        [Required]
        [StringLength(200)]
        public string CompanyName { get; set; }

        [StringLength(100)]
        public string Industry { get; set; }

        [Range(0, 1000)]
        public int YearsInBusiness { get; set; }

        [StringLength(5000)]
        public string Description { get; set; }

        [StringLength(3000)]
        public string Benefits { get; set; }

        [StringLength(500)]
        public string Address { get; set; }

        [StringLength(50)]
        public string Size { get; set; }

        [StringLength(200)]
        [Url]
        public string Website { get; set; }

        [StringLength(256)]
        [EmailAddress]
        public string ContactEmail { get; set; }

        [StringLength(20)]
        public string ContactPhone { get; set; }

        public int? LocationId { get; set; }

        public byte[]? Logo { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/requests/employer_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'employer_update_request.g.dart';

@JsonSerializable()
class EmployerUpdateRequest {
  final String companyName;
  final String? industry;
  final int? yearsInBusiness;
  final String? description;
  final String? benefits;
  final String? address;
  final String? size;
  final String? website;
  final String? contactEmail;
  final String? contactPhone;
  final int? locationId;
  final String? logo;

  EmployerUpdateRequest({
    required this.companyName,
    this.industry,
    this.yearsInBusiness,
    this.description,
    this.benefits,
    this.address,
    this.size,
    this.website,
    this.contactEmail,
    this.contactPhone,
    this.locationId,
    this.logo,
  });

  factory EmployerUpdateRequest.fromJson(Map<String, dynamic> json) => _$EmployerUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$EmployerUpdateRequestToJson(this);
}
